### ISSUE TITLE
Add support for RFC 8879 certificate compression

### ISSUE THREAD

**Issue by djc** [rustls/rustls#436]
Might be interesting to have, since it supposedly improves performance.

https://tools.ietf.org/html/rfc8879

**Comment by sharksforarms** [comment]
Submitted a WIP PR for this, let me know what you think! and I can make adjustments

**Comment by briansmith** [comment]
I understand the desire to avoid the compression library dependencies but also nobody would have any clue about how to use this API and so the benefit of adding this as-is would be very limited. I think if we want to do this then we should have (perhaps in a different crate) a very easy-to-use mechanism to actually use this with just a couple of lines of code.

Strawman:
* Add an rustls-compres-crate to the workspace of this repo for each compression algorithm we want to support.
* Only bother to support what's currently considered the best compression algorithm that is commonly used.
* Add compression support to the client and server demos that demonstrates which 1-3 lines of code to add to an application to get the compression support working.
* Document why we chose the algorithm(s) we did.
* Use pure Rust compression libraries without `unsafe` code, or if those aren't competitive performance-wise, explain the situation.
